src/Symfony/Component/DependencyInjection/Compiler/MergeExtensionConfigurationPass.php

```diff
@@ -121,3 +124,35 @@ public function getEnvPlaceholders()
         return null !== $this->processedEnvPlaceholders ? $this->processedEnvPlaceholders : parent::getEnvPlaceholders();
     }
 }
+
+/**
+ * A container builder preventing using methods that wouldn't have any effect from extensions.
+ *
+ * @internal
+ */
+class MergeExtensionConfigurationContainerBuilder extends ContainerBuilder
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function addCompilerPass(CompilerPassInterface $pass, $type = PassConfig::TYPE_BEFORE_OPTIMIZATION/*, int $priority = 0*/)
+    {
+        throw new LogicException(sprintf('You cannot add compiler passes to the container builder from an extension. Add it from the main container builder (from a bundle %s::build() method for instance).', Bundle::class));
```

Comment:
You cannot add compiler passes to the container builder from an extension. Add it from your bundle's "build()" method instead.